discovery-provider/src/tasks/index.py

```diff
@@ -391,10 +391,54 @@ def revert_blocks(self, db, revert_blocks_list):
         if rebuild_user_index:
             session.execute("REFRESH MATERIALIZED VIEW user_lexeme_dict")
 
+######## IPFS PEER REFRESH ########
+def refresh_peer_connections(task_context):
+    db = task_context.db
+    ipfs_client = task_context.ipfs_client
+    redis = task_context.redis
+    interval = int(task_context.shared_config["discprov"]["peer_refresh_interval"])
+    with db.scoped_session() as session:
+        db_cnode_endpts = (
+            session.query(
+                User.creator_node_endpoint).filter(
+                    User.creator_node_endpoint != None, User.is_current == True
+                ).distinct()
+        )
+
+        cnode_endpoints = {}
+        # Generate dictionary of unique creator node endpoints
+        for entry in db_cnode_endpts:
+            for cnode_user_set in entry:
+                cnode_entries = cnode_user_set.split(',')
+                for cnode_url in cnode_entries:
+                    if cnode_url == '':
+                        continue
+                    cnode_endpoints[cnode_url] = True
+
+        # Query ipfs information for each cnode endpoint
+        multiaddr_info = {}
+        for cnode_url in cnode_endpoints:
+            stored_in_redis = redis.get(cnode_url)
+            if stored_in_redis is not None:
+                continue
+            if cnode_url == '':
+                continue
+
+            try:
+                multiaddr_info[cnode_url] = get_ipfs_info_from_cnode_endpoint(cnode_url)
+            except Exception as e:  # pylint: disable=broad-except
+                # Handle error in retrieval by not peering this node
+                logger.warning('Error retrieving info for %s, %s', cnode_url, str(e))
+                multiaddr_info[cnode_url] = None
+
+        for key in multiaddr_info:
+            stored_url_val = redis.get(key)
```

Comment:
@hareeshnagaraj why do another redis.get here?  the value multiaddr_info isn't used

Reply:
multiaddr_info is directly used here to set values but i think it is unnecessary since the check is performed above